packages/core/src/components/menu/_common.scss

```diff
@@ -38,18 +40,62 @@ $dark-menu-background-color: $dark-gray4 !default;
   }
 
   .pt-dark & {
-    @include dark-menu-item($disabled-selector);
+    @include dark-menu-item($disabled-selector, $hover-selector);
   }
 }
 
-@mixin dark-menu-item($disabled-selector: ".pt-disabled") {
+@mixin dark-menu-item($disabled-selector: ".pt-disabled", $hover-selector: ":hover") {
   color: inherit;
 
+  &:not(#{$disabled-selector})#{$hover-selector} {
+    background: $dark-menu-item-color-hover;
+    color: inherit;
+  }
+
   &#{$disabled-selector} {
     color: $pt-dark-text-color-disabled;
   }
 }
 
+@mixin menu-item-intent($text-color: $pt-intent-text-colors) {
+  @each $intent, $color in $text-color {
+    &.pt-intent-#{$intent} {
+      color: $color;
+
+      &::before,
+      &::after,
+      .pt-menu-item-label {
+        color: $color;
```

Comment:
Changed the icon colors.

I don't wanna do it exactly the same because the usage requires a delicate mix of `.pt-disabled` in the `_menu.scss` file and I want to keep that logic there. The `map-get` in callout just does the same thing as using `$color`. We do use both in places - I think you're collating them already in another branch?